src/software/ai/passing/field_pitch_division.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+#include "software/geom/rectangle.h"
+#include "software/world/field.h"
+
+/**
+ * A FieldPitchDivision is abstraction around how we split up the field.
+ *
+ * Pitch divisions with zones outside of the field may lead to undefined
+ * behaviour. Overlapping zones are supported, but not recommended.
+ */
+template <class ZoneEnum>
+class FieldPitchDivision
```

Comment:
I'm curious about the rationale to have this implemented with a 1-18 enum instead of, for example, a 6x3 grid with 2 coordinates to denote a zone?

Reply:
Mostly to match up w/ how soccer analysis is done: https://footballbh.net/2019/02/04/the-use-of-zone-14-the-key-to-winning-football-matches/

Also we get nice compile time checks that plays/pass generator are respecting the pitch division